File: src/Shared/Logistics.Shared.Models/TruckDto.cs
namespace Logistics.Shared.Models;

public class TruckDto
{
    public string? Id { get; set; }
    public string? TruckNumber { get; set; }
    public AddressDto? CurrentLocation { get; set; }
    public double? CurrentLocationLat { get; set; }
    public double? CurrentLocationLong { get; set; }
    public IEnumerable<EmployeeDto> Drivers { get; set; } = new List<EmployeeDto>();
    public IEnumerable<LoadDto> Loads { get; set; } = new List<LoadDto>();
}

File: src/Client/Logistics.OfficeApp/src/app/core/models/truck/TruckDto.ts
import {AddressDto} from '../AddressDto';
import {EmployeeDto} from '../employee/EmployeeDto';
import {LoadDto} from '../load/LoadDto';

export interface TruckDto {
  id: string;
  truckNumber: string;
  drivers: EmployeeDto[];
  currentLocation?: AddressDto;
  currentLocationLat?: number;
  currentLocationLong?: number;
  loads?: LoadDto[];
}
